apps/mobile/src/screens/CreateGroup/CreateGroupReview.js

```diff
@@ -45,22 +48,63 @@ export const CreateGroupReview = React.forwardRef((_props, ref) => {
 
           if (data?.group) {
             trackWithConsent(AnalyticsEvents.GROUP_CREATED, {}, currentUser, !currentUser)
-            clearStore()
             changeToGroup(data.group.slug, { skipCanViewCheck: true })
+            clearStoreTimeoutRef.current = setTimeout(() => {
+              clearStore()
+            }, 5000)
           } else {
+            setIsSubmitting(false)
             setError('Group may have been created, but there was an error. Please contact Hylo support.', error)
           }
         } catch (e) {
+          setIsSubmitting(false)
           setError(e.message)
         }
       })()
     }
   }, [submit])
 
+  useEffect(() => {
+    return () => {
+      if (clearStoreTimeoutRef.current) {
```

Comment:
why all this stuff?

Reply:
Wanting to ensure that the clearStore function runs, even if the navigation action happens